source/common/event/dispatcher_impl.cc

```diff
@@ -37,6 +37,7 @@ DispatcherImpl::DispatcherImpl(Buffer::FactoryPtr&& factory)
 DispatcherImpl::~DispatcherImpl() {}
 
 void DispatcherImpl::clearDeferredDeleteList() {
+  ASSERT(run_tid_ == 0 || run_tid_ == Thread::Thread::currentThreadId());
```

Comment:
Question: I assume you added `run_tid_ == 0` since we do call functions before we call run(). Out of curiosity when does that happen? Also, I think we can add the asserts on all functions other than post() ?